src/App.tsx

```diff
@@ -51,12 +51,6 @@ class App extends Component {
     showJoinShareError: false
   };
 
-  // debounce so we don't send up partially-updated data contexts while syncing
-  writeDataContext = pDebounce(async (dataContext: DataContext | string | null) => {
-    const sharableDataContext = dataContext && await Codap.getSharableDataContext(dataContext);
-    sharableDataContext && database.set("dataContext", sharableDataContext);
-  }, 100);
-
```

Comment:
Even if we are now debouncing the DataContext response, is it not worth keeping the debouncing of the `writeDataContext`, simply so that we don't ever write to it too frequently on multiple changes? Or are we sure this can't happen any more?

Reply:
`writeDataContext()` is called from three places -- `initiateShare()`, `joinShare()`, and `handleDataContextUpdate()`. The bug that led me here in the first place was that in the case of `joinShare()` the sequence of events was essentially:

- call debounced `writeDataContext()` -- wait 100 ms
- call `writeUserItems()` -- actually write user items
- clients receive notification of new items
- actually write the new data context
- clients receive notification of updated context

The net effect was that clients did not process values for new attributes because they didn't have those attributes until after they received the items that required them. It seems to me that we always want to write immediately on initiate/join, and only debounce in the third case, when we're responding to CODAP notifications. If we encounter another client that wants to call `writeDataContext()` that should be debounced with the CODAP response, we can introduce a `debouncedWriteDataContext()` at that point.